commandapi-platforms/commandapi-spigot/commandapi-spigot-core/src/main/java/dev/jorel/commandapi/CommandAPISpigot.java

```diff
@@ -0,0 +1,136 @@
+import dev.jorel.commandapi.commandsenders.BukkitRemoteConsoleCommandSender;
+import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
+import dev.jorel.commandapi.nms.NMS;
+import dev.jorel.commandapi.nms.SpigotNMS;
+import dev.jorel.commandapi.wrappers.NativeProxyCommandSender;
+import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.Bukkit;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandMap;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+import org.bukkit.command.ProxiedCommandSender;
+import org.bukkit.command.RemoteConsoleCommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerLoadEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public abstract class CommandAPISpigot<Source> extends CommandAPIBukkit<Source> implements SpigotNMS<Source> {
+
+	private static CommandAPISpigot<?> spigot;
+
+	@SuppressWarnings("unchecked")
+	protected CommandAPISpigot() {
+		this.nms = (NMS<Source>) bukkitNMS();
+		CommandAPISpigot.spigot = this;
+		setInstance(this);
+	}
+
+	@SuppressWarnings("unchecked")
+	public static <Source> CommandAPISpigot<Source> getSpigot() {
+		if (spigot != null) {
+			return (CommandAPISpigot<Source>) spigot;
+		}
+		throw new IllegalStateException("Tried to access CommandAPIBukkit instance, but it was null! Are you using CommandAPI features before calling CommandAPI#onLoad?");
+	}
+
+	public static InternalSpigotConfig getConfiguration() {
+		return (InternalSpigotConfig) CommandAPIBukkit.getConfiguration();
+	}
+
+	private static void setInternalConfig(InternalSpigotConfig config) {
+		CommandAPIBukkit.config = config;
+	}
+
+	@Override
+	public void onLoad(CommandAPIConfig<?> config) {
+		if (config instanceof CommandAPISpigotConfig spigotConfig) {
+			CommandAPISpigot.setInternalConfig(new InternalSpigotConfig(spigotConfig));
+		} else {
+			CommandAPI.logError("CommandAPIBukkit was loaded with non-Bukkit config!");
+			CommandAPI.logError("Attempts to access Bukkit-specific config variables will fail!");
+		}
+		super.onLoad();
+	}
+
+	/**
+	 * Enables the CommandAPI. This should be placed at the start of your
+	 * <code>onEnable()</code> method.
+	 */
+	public static void onEnable() {
+		CommandAPIBukkit.get().plugin = getConfiguration().getPlugin();
+
+		Bukkit.getScheduler().scheduleSyncDelayedTask(CommandAPIBukkit.get().plugin, () -> {
+			// Sort out permissions after the server has finished registering them all
+			CommandAPIBukkit.get().getCommandRegistrationStrategy().runTasksAfterServerStart();
+			if (!getConfiguration().skipReloadDatapacks()) {
+				CommandAPIBukkit.get().reloadDataPacks();
+			}
+			CommandAPIBukkit.get().updateHelpForCommands(CommandAPI.getRegisteredCommands());
+		}, 0L);
+
+		CommandAPIBukkit.get().stopCommandRegistrations();
```

Comment:
`CommandAPISpigot` extends `CommandAPIBukkit`, so I don't think `CommandAPIBukkit.get()` is needed anymore. Same in Paper.